pkg/registry/apis/provisioning/jobs/migrate/legacy_resources.go

```diff
@@ -70,17 +71,29 @@ func (m *legacyResourcesMigrator) Migrate(ctx context.Context, rw repository.Rea
 	}
 
 	progress.SetMessage(ctx, "migrate folders from SQL")
-	if err := m.folderMigrator.Migrate(ctx, namespace, repositoryResources, progress); err != nil {
-		return fmt.Errorf("migrate folders from SQL: %w", err)
+	if m.clients != nil {
```

Comment:
adding this nil check is a hack for the tests but not great for the code / logic :)